application.rb

```diff
@@ -189,9 +201,10 @@ def legacy_convert params
   private
 
   def authenticate!
+    # Allow authentication via Header token so we can do JSON API requests
+    return true if request.env["HTTP_ASQ_API_TOKEN"] == ENV["ASQ_API_TOKEN"]
```

Comment:
This isn't secure enough for my taste. Please find a more secure way of validating for the API call.